zrml/prediction-markets/src/migrations.rs

```diff
@@ -337,6 +349,99 @@ impl<T: Config> OnRuntimeUpgrade for CleanUpStorageForResolvedOrClosedMarkets<T>
+
+        let market_ids_per_dispute_iterator: IterType<T> =
+            frame_support::migration::storage_key_iter::<_, _, Twox64Concat>(
+                b"PredictionMarkets",
+                b"MarketIdsPerDisputeBlock",
+            );
+
+        let market_ids_per_dispute: Vec<DisputeBlockToMarketIdsTuple<T>> =
+            market_ids_per_dispute_iterator.collect();
+        for (dispute_start_block, market_ids) in market_ids_per_dispute {
+            total_weight = total_weight.saturating_add(T::DbWeight::get().writes(2));
+            // NOTE: These migration only makes sense on BS runtime so its fine to assume
+            // DisputePeriod equal to BLOCKS_PER_DAY
+            let dispute_period: T::BlockNumber = BLOCKS_PER_DAY.saturated_into::<u32>().into();
+            let new_dispute_start_block = dispute_start_block.saturating_add(dispute_period);
+            MarketIdsPerDisputeBlock::<T>::insert(new_dispute_start_block, market_ids);
+            MarketIdsPerDisputeBlock::<T>::remove(dispute_start_block);
+        }
+
+        let market_ids_per_report_iterator: IterType<T> =
+            frame_support::migration::storage_key_iter::<_, _, Twox64Concat>(
+                b"PredictionMarkets",
+                b"MarketIdsPerReportBlock",
+            );
+
+        let market_ids_per_report: Vec<_> = market_ids_per_report_iterator.collect();
+        for (dispute_start_block, market_ids) in market_ids_per_report {
+            total_weight = total_weight.saturating_add(T::DbWeight::get().writes(2));
+            // NOTE: These migration only makes sense on BS runtime so its fine to assume
+            // DisputePeriod equal to BLOCKS_PER_DAY
+            let dispute_period: T::BlockNumber = BLOCKS_PER_DAY.saturated_into::<u32>().into();
+            let new_dispute_start_block = dispute_start_block.saturating_add(dispute_period);
+            MarketIdsPerReportBlock::<T>::insert(new_dispute_start_block, market_ids);
+            MarketIdsPerReportBlock::<T>::remove(dispute_start_block);
+        }
+        StorageVersion::new(PREDICTION_MARKETS_NEXT_STORAGE_VERSION_FOR_MIGRATE_MARKET_IDS_STORAGE)
+            .put::<Pallet<T>>();
+        total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+
+        log::info!("Completed storage migration of MigrateMarketIdsPerBlockStorage");
+        total_weight
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn pre_upgrade() -> Result<(), &'static str> {
+        Ok(())
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn post_upgrade() -> Result<(), &'static str> {
+        let dispute_period: T::BlockNumber = BLOCKS_PER_DAY.saturated_into::<u32>().into();
+        for (key, market_ids) in MarketIdsPerDisputeBlock::<T>::iter() {
+            for market_id in market_ids {
+                let disputes = Disputes::<T>::get(&market_id);
+                let dispute = disputes.last().ok_or("No dispute found")?;
+                assert!(key == dispute.at + dispute_period)
```

Comment:
This should be done using `assert_eq!`, and should contain a message in case of error so you'll have an easier time debugging.